src/distributed-transaction/percolator.md

```diff
@@ -0,0 +1,156 @@
+# Percolator
+
+TiKV supports distributed transactions, which is inspired by Google's [Percolator](https://ai.google/research/pubs/pub36726.pdf). In this section, we will briefly introduce Percolator and how we make use of it in TiKV.
+
+## What is Percolator?
+
+*Percolator* is a system built by Google for incremental processing on a very large data set. Since this is just a brief introduction, you can view the full paper [here](https://ai.google/research/pubs/pub36726#) for more details. If you are already very familiar with it, you can skip this section and go directly to [Percolator in TiKV](#Percolator-in-TiKV)
+
+Percolator is built based on Google's BigTable, a distributed storage system that supports single-row transactions. Percolator implements distributed transactions in ACID snapshot-isolation semantics, which is not supported by BigTable. A column `c` of Percolator is actually divided into the following internal columns of BigTable:
+
+* `c:lock`
+* `c:write`
+* `c:data`
+* `c:notify`
+* `c:ack_O`
+
+Percolator also relies on a service named *timestamp oracle*. The timestamp oracle can produce timestamps in a strictly increasing order. All read and write operations need to apply for timestamps from the timestamp oracle, and a timestamp coming from the timestamp oracle will be regarded as the logical time when the read/write operation happens.
+
+Percolator is a multi-version storage, and a data item's version is represented by the timestamp when the transaction was committed.
+
+For example, in such a state:
+
+| key | v:data | v:lock | v"write |
+|-----|--------|--------|---------|
+|k1   |14:"value2"<br/>12:<br/>10:"value1"|14:primary<br/>12:<br/>10:|14:<br/>12:data@10<br/>10:|
+
+The state in the table means that for key `k1`, value `"value1"` was committed at timestamp `12`. Then there is an uncommitted version whose value is `"value2"`, and it's uncommitted because there's a lock. You will understand why it is like this after understanding how transactions work.
+
+The data of a specific row and a specific Percolator column is named a *cell*. The above example shows different versions of data for a single cell.
```

Comment:
```suggestion
The data of a specific row and the corresponding column in Percolator is named a *cell*. The above example shows different versions of data for a single cell.
```

Reply:
The column is not the corresponding thing to the row. A *cell* is of a row and is also of a Percolator's column. Is there any better expression?